Orange/data/domain.py

```diff
@@ -58,69 +69,101 @@ def __init__(self, source, destination):
 
 
 class Domain:
-    def __init__(self, attributes, class_vars=None, metas=None, source=None):
+    def __init__(self, attributes=(), class_vars=(), metas=(), source=None):
```

Comment:
Specifying some variables as strings and others as variables does not make much sense in, say, widgets. Let `from_domain` be strict. However, this form of constructor is useful for lecturing since it allows you to refer to attributes by names when constructing new domains (and to also add new, computed attributes to the existing domain) in the command line, all without too much typing. So please leave this one as it is, but I sure am not against a stricter `from_domain`, either.